core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1014,6 +1014,15 @@ package object config {
       .timeConf(TimeUnit.NANOSECONDS)
       .createWithDefaultString("1s")
 
+  private[spark] val LISTENER_BUS_EVENT_QUEUE_WAIT_FOR_EVENT_DISPATCH_EXIT_ON_STOP =
+    ConfigBuilder("spark.scheduler.listenerbus.eventqueue.waitForEventDispatchExitOnStop")
+      .doc("Whether wait until the dispatch thread exit when stop invoked. " +
+        "This is set to true by default for graceful shutdown of the event queue, " +
+        "but allow user to configure the behavior if they don't need.")
+      .version("4.0.0")
+      .booleanConf
+      .createWithDefault(true)
```

Comment:
Instead of making this a Boolean, I would suggest setting this as an int instead - and set default to 0 to preserve existing behavior.

Reply:
Hi @mridulm I checked other configurations, seems like Boolean type are used in the type of conf like enable/disable something, so I would like to follow the style. Here is one example:
`
  private[spark] val LISTENER_BUS_LOG_SLOW_EVENT_ENABLED =
    ConfigBuilder("spark.scheduler.listenerbus.logSlowEvent")
      .internal()
      .doc("When enabled, log the event that takes too much time to process. This helps us " +
        "discover the event types that cause performance bottlenecks. The time threshold is " +
        "controlled by spark.scheduler.listenerbus.logSlowEvent.threshold.")
      .version("3.0.0")
      .booleanConf
      .createWithDefault(true)
`

Reply:
To clarify, I am suggesting to use int to represent a boolean.
Rather, use it to set the wait time - where `0`, the default value, will mean wait until stopped - which preserves the current behavior.

Reply:
@mridulm Got it, thanks for your suggestion, I just commit the change, please have a check.